Context canceling causing browser integration tests to fail very often

### Brief summary

Run the browser integration tests in ``. Almost 99% of the time one of the tests will fail with:

```shell
time="2025-12-16T14:22:55Z" level=error msg="error on evaluating window.k6SpanId: evaluating JS in global context: context canceled" category="FrameSession:onFrameNavigated" elapsed="0 ms" source=browser
panic: GoError: browser is attaching to target: canceled

goroutine 3551 [running]:
go.k6.io/k6/js/common.Throw(...)
	/home/runner/work/k6/k6/js/common/util.go:17
go.k6.io/k6/internal/js/modules/k6/browser/k6ext.Panicf.func1(0xc0000e7808, {0xc0034628c0, 0x1, 0x1})
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/k6ext/panic.go:35 +0x9c
go.k6.io/k6/internal/js/modules/k6/browser/k6ext.sharedPanic({0x23a7b20, 0xc002caa690}, 0xc001887e38, {0xc0034628c0, 0x1, 0x1})
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/k6ext/panic.go:76 +0x39c
go.k6.io/k6/internal/js/modules/k6/browser/k6ext.Panicf({0x23a7b20, 0xc002caa690}, {0x20cf8a3, 0x22}, {0xc0034628c0, 0x1, 0x1})
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/k6ext/panic.go:37 +0xb1
go.k6.io/k6/internal/js/modules/k6/browser/common.(*Browser).initEvents.func1()
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/common/browser.go:238 +0x54b
created by go.k6.io/k6/internal/js/modules/k6/browser/common.(*Browser).initEvents in goroutine 172
	/home/runner/work/k6/k6/internal/js/modules/k6/browser/common/browser.go:219 +0x24b
FAIL	go.k6.io/k6/internal/js/modules/k6/browser/tests	11.988s
```

### k6 version

NA

### OS

NA

### Docker version and image (if applicable)

_No response_

### Steps to reproduce the problem

NA

### Expected behaviour

Retry the connection, and possibly ignore the context in CI

### Actual behaviour

Fails in CI

--- a/internal/js/modules/k6/browser/common/browser.go
+++ b/internal/js/modules/k6/browser/common/browser.go
@@ -234,11 +234,16 @@ func (b *Browser) initEvents() error {
 			case event := <-chHandler:
 				if ev, ok := event.data.(*target.EventAttachedToTarget); ok {
 					b.logger.Debugf("Browser:initEvents:onAttachedToTarget", "sid:%v tid:%v", ev.SessionID, ev.TargetInfo.TargetID)
-					if err := b.onAttachedToTarget(ev); err != nil {
-						if errors.Is(err, context.Canceled) ||
-							errors.Is(err, context.DeadlineExceeded) {
-							continue
-						}
+					err := b.onAttachedToTarget(ev)
+					if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+						b.logger.Debugf(
+							"Browser:initEvents:onAttachedToTarget:suppressed",
+							"sid:%v tid:%v err:%v",
+							ev.SessionID, ev.TargetInfo.TargetID, err,
+						)
+						continue
+					}
+					if err != nil {
 						k6ext.Panicf(b.vuCtx, "browser is attaching to target: %w", err)
 					}
 				} else if ev, ok := event.data.(*target.EventDetachedFromTarget); ok {

